core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -777,6 +778,42 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * This method compresses the files passed in, and writes the compressed data out into the
+   * [[OutputStream]] passed in. Each file is written as a new [[ZipEntry]] with its name being
+   * the name of the file being compressed.
+   */
+  private[spark] def zipFilesToStream(
+      files: Seq[Path],
+      hadoopConf: Configuration,
+      outputStream: OutputStream): Unit = {
+    val fs = FileSystem.get(hadoopConf)
+    val buffer = new Array[Byte](64 * 1024)
+    val zipStream = Some(new ZipOutputStream(outputStream))
+    try {
+      files.foreach { remotePath =>
+        val inputStream = Some(fs.open(remotePath, 1 * 1024 * 1024)) // 1MB Buffer
```

Comment:
I don't think you gain anything by wrapping the zipStream & inputStream in a `Some`